src/libraries/Microsoft.Extensions.DependencyInjection/tests/NativeAotTests/Microsoft.Extensions.DependencyInjection.NativeAotTests.proj

```diff
@@ -0,0 +1,14 @@
+<Project DefaultTargets="Build">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove(Directory.Build.props))" />
+
+  <PropertyGroup>
+    <AdditionalProjectReferences>Microsoft.Extensions.DependencyInjection</AdditionalProjectReferences>
+  </PropertyGroup>
+  
+  <ItemGroup>
+    <TestConsoleAppSourceFiles Include="..\TrimmingTests\ActivatorUtilitiesTests.cs" />
+    <TestConsoleAppSourceFiles Include="..\TrimmingTests\ServiceCollectionExtensionsTests.cs" />
```

Comment:
What's the key by which we decide the trimming tests that should be also native AOT tests?

ServiceCollectionExtensionsTests would not have hit the issue we were discussing and feels as applicable as any other trimming test.

Should we just run all trimming tests? There's a greater than zero chance they may catch _something_ and it only makes the CI a couple minutes slower.

These tests are testing assumptions under which we suppressed static analysis warnings. We've made the wrong call on suppressing a warning many times in the past. There could be an AOT angle to the trimming suppression. I think that might tip the scale to "all TrimmingTests should be AOT tests".

Reply:
> Should we just run all trimming tests?

Yes, I think that would be a good idea. I don't have time to do it myself right now, so it would be great if someone else could pick that up.

Reply:
> > Should we just run all trimming tests?
> 
> Yes, I think that would be a good idea. I don't have time to do it myself right now, so it would be great if someone else could pick that up.

Seemed straightforward enough so I just did it in #101229. Looks like it found stuff. Unclear if test issues or product issues at this point.